libecole/src/observation/pseudocosts.cpp

```diff
@@ -0,0 +1,52 @@
+#include <cmath>
+#include <cstddef>
+
+#include <scip/scip_var.h>
+#include <scip/scipdefplugins.h>
+#include <scip/utils.hpp>
+
+#include "ecole/observation/pseudocosts.hpp"
+#include "ecole/scip/model.hpp"
+#include "ecole/scip/type.hpp"
+
+namespace ecole {
+namespace observation {
+
+nonstd::optional<xt::xtensor<double, 1>> Pseudocosts::obtain_observation(scip::Model& model) {
+
+	if (model.get_stage() == SCIP_STAGE_SOLVING) {
+
+		SCIP* scip = model.get_scip_ptr();
+
+		SCIP_VAR** cands;
+		SCIP_Real* cands_lp_values;
+		int nb_cands;
+
+		/* get branching candidates */
+		SCIPgetLPBranchCands(scip, &cands, &cands_lp_values, NULL, NULL, &nb_cands, NULL);
+		assert(nb_cands > 0);
+
+		/* Store pseudocosts in tensor */
+		auto const nb_lp_columns = static_cast<std::size_t>(SCIPgetNLPCols(scip));
+		auto pseudocosts = xt::xarray<double>::from_shape({nb_lp_columns});
+		pseudocosts.fill(std::nan(""));
```

Comment:
- I believe it should be a `xt::xtensor`
- I believe you can use in one shot initialization and filling with
   ```cpp
   xt::xtensor<double, 1> pseudocosts({nb_lp_columns}, std::nan(""));
   ```

Reply:
This is from
https://github.com/ds4dm/ecole/blob/727af527279a23e42c8f061fccdc0465a390523e/libecole/src/observation/strongbranchingscores.cpp#L68
which code you previously accepted and merged into master.

Reply:
I understand, but my reviews are not perfect. Every new ones bring new opportunity for improvement.

Reply:
Ahah yeah I know :-) I was more saying, we should probably do similar changes in the strong branching code too then. Do you want me to incorporate that (fixes to SB code) in this pull request as well?

Reply:
That would be great! Perhaps make it a separate commit though?